### ISSUE TITLE
named condition gives unexpected result

### ISSUE THREAD

**Issue by edzer** [r-lib/testthat#751]
reprex: running
```r
test_that("bla", {
    cond = TRUE
    names(cond) = "foo"
    skip_if_not(cond)
    print("foo!")
})
```
doesn't print `foo`.

**Comment by vincentvanhees** [comment]
You probably want to use cat() instead of print()

**Comment by hadley** [comment]
We rely on R's definition of `isTRUE()` so this is fixed in R 3.5
